internal/v2/controller/http/callback.go

```diff
@@ -0,0 +1,120 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+//
+// Copyright (C) 2020 IOTech Ltd
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package http
+
+import (
+	"encoding/json"
+	"net/http"
+
+	sdkCommon "github.com/edgexfoundry/device-sdk-go/internal/common"
+	"github.com/edgexfoundry/device-sdk-go/internal/v2/application"
+	edgexErr "github.com/edgexfoundry/go-mod-core-contracts/errors"
+	contract "github.com/edgexfoundry/go-mod-core-contracts/models"
+	"github.com/gorilla/mux"
+)
+
+func (c *V2HttpController) DeleteDevice(writer http.ResponseWriter, request *http.Request) {
+	vars := mux.Vars(request)
+	id := vars[sdkCommon.IdVar]
+	correlationID := request.Header.Get(sdkCommon.CorrelationHeader)
+
+	err := application.DeleteDevice(id, c.dic)
+	if err == nil {
+		c.sendResponse(writer, request, sdkCommon.APIV2DeviceCallbackIdRoute, nil, http.StatusOK)
+	} else {
+		c.sendEdgexError(writer, request, err, sdkCommon.APIV2DeviceCallbackIdRoute, correlationID)
+	}
+}
+
+func (c *V2HttpController) AddUpdateDevice(writer http.ResponseWriter, request *http.Request) {
+	defer request.Body.Close()
+	correlationID := request.Header.Get(sdkCommon.CorrelationHeader)
+
+	// TODO: change to v2dto when core services are ready.
+	var device contract.Device
+	err := json.NewDecoder(request.Body).Decode(&device)
+	if err != nil {
+		c.sendError(writer, request, edgexErr.KindServerError, "failed to decode JSON", err, sdkCommon.APIV2DeviceCallbackRoute, correlationID)
+		return
+	}
+
+	isAdd := request.Method == http.MethodPost
+	edgexErr := application.AddUpdateDevice(isAdd, device, c.dic)
+	if edgexErr == nil {
+		c.sendResponse(writer, request, sdkCommon.APIV2DeviceCallbackRoute, nil, http.StatusOK)
+	} else {
+		c.sendEdgexError(writer, request, edgexErr, sdkCommon.APIV2DeviceCallbackRoute, correlationID)
+	}
+}
+
+func (c *V2HttpController) DeleteProfile(writer http.ResponseWriter, request *http.Request) {
+	vars := mux.Vars(request)
+	id := vars[sdkCommon.IdVar]
+	correlationID := request.Header.Get(sdkCommon.CorrelationHeader)
+
+	err := application.DeleteProfile(id, c.dic)
+	if err == nil {
+		c.sendResponse(writer, request, sdkCommon.APIV2ProfileCallbackIdRoute, nil, http.StatusOK)
+	} else {
+		c.sendEdgexError(writer, request, err, sdkCommon.APIV2ProfileCallbackIdRoute, correlationID)
+	}
+}
+
+func (c *V2HttpController) AddUpdateProfile(writer http.ResponseWriter, request *http.Request) {
+	defer request.Body.Close()
+	correlationID := request.Header.Get(sdkCommon.CorrelationHeader)
+
+	// TODO: change to v2dto when core services are ready.
+	var profile contract.DeviceProfile
```

Comment:
This is a V2 API which should be receiving a Request DTO, which are here: https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/v2/dtos/requests/deviceprofile.go#L22

Looks like the Swagger needs to be updated to use `AddDeviceProfileRequest` and `UpdateDeviceProfileRequest` and this code needs to handle the request objects separately. For core services updates are not handled on a field by field basis rather then delete and replace. Should be able to do same in the SDK and just update the current profile.

Reply:
@lenny-intel there are more work in the SDK internal work, such as handling V2 model in cache.  I thought we can implement the http controller layer in this PR, and open another issue to track the application layer for V2 callback.

Reply:
@cloudxxx8, So what you are saying is the V2 API initially is receiving V1 models? 

I get the issue with having the V2 models in the cache. What about a temporary conversion of the V2 model to V1 model as it is put into the cache? 

This way we just remove the conversion when the cache is ready for V2 models.

Reply:
after discussing with Chris, we decided to implement V2 cache first.  The conversion might spend the same time as developing V2 cahce